flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperator.java

```diff
@@ -261,6 +274,10 @@ public void run() {
 
 							if (currentSplit.equals(EOS)) {
 								isRunning = false;
+								if (this.isFormatOpen) {
```

Comment:
is this necessary? Wouldn't we directly go into the `finally` clause because of the `break` where the IF is closed anyway?